README.rst

```diff
@@ -72,6 +51,96 @@ and avoid using an external process::
         f.write(b"Hello")
 
 
+The ``xopen`` function
+----------------------
+
+The ``xopen`` module offers a single function named ``xopen`` with the following
+signature::
+
+  xopen(
+    filename: str | bytes | os.PathLike,
+    mode: Literal["r", "w", "a", "rt", "rb", "wt", "wb", "at", "ab"] = "r",
+    compresslevel: Optional[int] = None,
+    threads: Optional[int] = None,
+    *,
+    encoding: str = "utf-8",
+    errors: Optional[str] = None,
+    newline: Optional[str] = None,
+    format: Optional[str] = None,
+  ) -> IO
+
+The function opens the file using a function suitable for the detected
+file format and returns an open file-like object.
+
+When writing, the file format is chosen based on the file name extension:
+``.gz``, ``.bz2``, ``.xz``, ``.zst``. This can be overriden with ``format``.
+If the extension is not recognized, no compression is used.
+
+When reading and a file name extension is available, the format is detected
+from the extension.
+When reading and no file name extension is available,
+the format is detected from the contents.
+
+Parameters
+~~~~~~~~~~
+
+**filename** (str, bytes, or `os.PathLike <https://docs.python.org/3/library/os.html#os.PathLike>`_):
+Name of the file to open.
+
+If set to ``"-"``, standard output (in mode ``"w"``) or
+standard input (in mode ``"r"``) is returned.
+
+**mode**, **encoding**, **errors**, **newline**:
+These parameters have the same meaning as in Python’s built-in
+`open function <https://docs.python.org/3/library/functions.html#open>`_
+except that the default encoding is always UTF-8 instead of the
+preferred locale encoding.
+``encoding``, ``errors`` and ``newline`` are only used when opening a file in text mode.
+
+**compresslevel**:
+The compression level for writing to gzip, xz and Zstandard files.
+If set to None, a default depending on the format is used:
+gzip: 1, xz: 6, Zstandard: 3.
+
+This parameter is ignored for other compression formats.
+
+**format**:
+Override the autodetection of the input or output format.
+Possible values are: ``"gz"``, ``"xz"``, ``"bz2"``, ``"zst"``.
+
+**threads**:
+If multi-threaded compression or decompression is available,
+this parameter can be used to override the number of threads
+used. It is ignored otherwise.
+
+Set threads to 0 to force opening the file without using a subprocess.
+For some compression levels,
+compressed files are by default read or written
+using a pipe to a subprocess running an external tool such as
+``pbzip2`` or ``xz``.
+With *threads* set to 0, a normal function call is used instead.
+
+
+Backends
+--------
+
+Opening of gzip files is delegated to one of these programs or libraries:
+
+* `python-isal <https://github.com/pycompression/python-isal>`_.
+  Supports multiple threads and compression levels up to 3.
+* zlib-ng
```

Comment:
```
* `python-zlib-ng <https://github.com/pycompression/python-zlib-ng>`_. 
  Supports multiple threads and compression levels up to 9.
```

Reply:
Whoops, I wrote this while our broadband was updated and had no internet access. I had intended meant to add the link later.

Reply:
BTW, if you make these comments on the 'Files changed' tab, then you can use the 'Suggestions' button and I can just click to accept them.

See point 6 at https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/commenting-on-a-pull-request#adding-comments-to-a-pull-request

Reply:
Ah, that is good to know. Thanks!